app/containers/WeatherContainer/index.js

```diff
@@ -0,0 +1,263 @@
+      setLoading(false);
+    }
+  }, [cityData]);
+
+  useEffect(() => {
+    if (cityName && !cityData) {
+      dispatchCityWheather(cityName);
+      setLoading(true);
+    }
+  }, []);
+
+  const handleOnChange = wName => {
+    if (!isEmpty(wName)) {
+      dispatchCityWheather(wName);
+      setLoading(true);
+    }
+  };
+  const debouncedHandleOnChange = debounce(handleOnChange, 400);
+
+  const renderCityList = () => {
+    return (
+      (cityData || loading) && (
+        <CustomCard>
+          <Skeleton loading={loading} active>
+            {cityName && <div></div>}
+
+            <CustomCard>
+              <h2>City Name - {cityData.name}</h2>
+              <h2>City Timezone - {cityData.timezone}</h2>
+              <h2>City Temperature - {cityData.main.temp}</h2>
+
+              {/* <h2 id="rpository_name" values={{ name: cityData.name }} />
+              <h2 id="repository_full_name" values={{ fullName: cityData.temp_min }} /> */}
+            </CustomCard>
+          </Skeleton>
+        </CustomCard>
+      )
+    );
+  };
+  const renderErrorState = () => {
+    let citywError;
+    if (cityError) {
+      citywError = cityError;
+    }
+    return (
+      !cityData &&
+      citywError && (
+        // eslint-disable-next-line react/prop-types
+        <h2>Error</h2>
+      )
+    );
+  };
+  const refreshPage = () => {
+    history.push('stories');
+    window.location.reload();
+  };
+  return (
+    <Container>
+      <RightContent>
+        <Clickable textId="stories" onClick={refreshPage} />
+      </RightContent>
+      <CustomCard title="hey">
+        <T marginBottom={10} id="get_repo_details" />
+        <Search
+          data-testid="search-bar"
+          defaultValue={cityName}
+          type="text"
+          onChange={evt => debouncedHandleOnChange(evt.target.value)}
+          onSearch={searchText => debouncedHandleOnChange(searchText)}
+        />
+      </CustomCard>
+      {renderCityList()}
+      {renderErrorState()}
+    </Container>
+  );
+}
+
+WeatherContainer.propTypes = {
+  dispatchCityWheather: PropTypes.func,
+  intl: PropTypes.object,
+  cityData: PropTypes.object,
+  cityError: PropTypes.object,
+  cityName: PropTypes.string,
+  history: PropTypes.object,
+  maxwidth: PropTypes.number,
+  padding: PropTypes.number
+};
+
+const mapStateToProps = createStructuredSelector({
+  weatherContainer: selectWeatherContainer(),
+  cityData: selectCityData(),
+  cityError: selectCityError(),
+  cityName: selectCityName()
+});
+
+function mapDispatchToProps(dispatch) {
+  const { requestGetCityWeather } = weatherContainerCreators;
+  return {
+    dispatchCityWheather: cityName => dispatch(requestGetCityWeather(cityName))
+  };
+}
+
+const withConnect = connect(
+  mapStateToProps,
+  mapDispatchToProps
+);
+
+export default compose(
+  withConnect,
+  memo
+)(WeatherContainer);
+
+export const WeatherContainerTest = compose(injectIntl)(WeatherContainer);
+
+//   return (
```

Comment:
i'm sure you'll probably be cleaning this up in the next commit but just so that we're on the same page, we want to avoid having commented out code as far as possible